### PR TITLE
Replace url to uri for resolving schema location

### PR DESCRIPTION
**Pull request by uzairbaig**
Replace URLs to URIs for resolving schema location.

### CODE REVIEW

**Comment by crossing** [comment]
Just a comment, conceptually URIs do not indicate how to retrieve a resource or where a resource is, but URLs do. If it's expected to retrieve the schemas from these locations, URLs are better fit.

File: catalog-core/src/main/java/uk/gov/justice/schema/catalog/FileContentsAsStringLoader.java [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 import java.io.IOException;
-import java.net.URL;
+import java.net.URI;
 
 import org.apache.commons.io.IOUtils;
 
 public class FileContentsAsStringLoader {
 
-    public String readFileContents(final URL urlOfFile) {
+    public String readFileContents(final URI urlOfFile) {
```
Comment (mapingo) [inline catalog-core/src/main/java/uk/gov/justice/schema/catalog/FileContentsAsStringLoader.java]: urlOfFile argument can be changed to uriOfFile

File: catalog-core/src/test/java/uk/gov/justice/schema/catalog/CatalogToSchemaResolverTest.java [hunk 2]
```diff
@@ -43,9 +42,9 @@ public void shouldMapSchemasFoundOnTheClasspathToTheirIds() throws Exception {
         final Group group = mock(Group.class);
         final Schema schema = mock(Schema.class);
 
-        final URL url = new URL("file:/src/main/schema.json");
+        final URI url = new URI("file:/src/main/schema.json");
```
Comment (mapingo) [inline catalog-core/src/test/java/uk/gov/justice/schema/catalog/CatalogToSchemaResolverTest.java]: url can be renamed to uri

File: catalog-core/src/test/java/uk/gov/justice/schema/catalog/SchemaResolverTest.java [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 import uk.gov.justice.schema.catalog.util.UrlConverter;
```
Comment (mapingo) [inline catalog-core/src/test/java/uk/gov/justice/schema/catalog/SchemaResolverTest.java]: I am guessing we may longer need this import...the UrlConverter class probably can be deleted from the code base now

**Comment by purple52** [comment]
I agree with @crossing - we need to be clear why we're using URIs vs URLs in each place in the code. Some of these should probably remain as URLs.

**Pull request merged**